### PR TITLE
[Peering Service] apiVersion:2020-01-01-preview 

### PR DESCRIPTION
**Pull request by rileymckenna**
Added Registered Asn
Added Legacy Peering filter by Asn
Added filter to peering service countries
Added Sku to peering type
Added type to direct peering enum
Changed how contact details are stored
Added Sku to peering service

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "2019-09-01-preview",
+    "version": "2020-01-01-preview",
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: >"2020-0 [](start = 15, length = 7)

revert to 2019

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 2]
```diff
@@ -328,46 +336,521 @@
...
+            "description": "The type of direct peering.",
+            "required": false,
+            "type": "string",
+            "enum": [
+              "Edge",
+              "Transit",
+              "Cdn",
+              "Internal",
+              "Ix",
+              "IxRs"
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: this is not modelled as a string. So, it is a breaking change to existing clients/SDK/CLIs...

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 3]
```diff
@@ -328,46 +336,521 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredAsns": {
+      "get": {
+        "tags": [
+          "PeeringRegisteredAsns"
+        ],
+        "description": "Lists all registered ASNs under the given subscription, resource group and peering.",
+        "operationId": "RegisteredAsns_ListByPeering",
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: >RegisteredAsns_ListByPeering [](start = 24, length = 28)

also add _ListBySubscription and _ListByResourceGroup

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 4]
```diff
@@ -328,46 +336,521 @@
...
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: is it long running ?

If so, please mark  "x-ms-long-running-operation": true,  
Update this for all 201/201 LROs.

Also, to let SDK know how to track.. add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 5]
```diff
@@ -328,46 +336,521 @@
...
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/{peeringName}/registeredPrefixes": {
       "get": {
         "tags": [
-          "PeeringLocations"
+          "PeeringRegisteredPrefixes"
         ],
-        "description": "Lists all of the available peering locations for the specified kind of peering.",
-        "operationId": "PeeringLocations_List",
+        "description": "Lists all registered prefixes under the given subscription, resource group and peering.",
+        "operationId": "RegisteredPrefixes_ListByPeering",
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: >RegisteredPrefixes_ListByPeering [](start = 24, length = 32)

add ListBySubscription and ListByRresourceGroup

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 6]
```diff
@@ -709,14 +1189,60 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/peeringServiceCountries": {
+      "get": {
+        "tags": [
+          "PeeringServiceCountries"
+        ],
+        "description": "Lists all of the available countries for peering service.",
+        "operationId": "PeeringServiceCountries_List",
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: >PeeringServiceCountries_List [](start = 24, length = 28)

PeeringServiceCountries_ListBySubscription

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 7]
```diff
@@ -1871,9 +2400,12 @@
...
-          "$ref": "#/definitions/ContactInfo",
-          "description": "The contact information of the peer."
+        "peerContactDetail": {
+          "description": "The contact details of the peer.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContactDetail"
+          }
         },
         "peerName": {
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: adding new property is a breaking change.. this is also removing.. please limit the these to new api version

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json [hunk 8]
```diff
@@ -2383,6 +3122,10 @@
         }
       ],
       "properties": {
+        "sku": {
+          "$ref": "#/definitions/PeeringServiceSku",
+          "description": "The SKU that defines the type of the peering service."
+        },
```
Comment (sanjaiganesh) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: adding new property.. breaking change..  [https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo](https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=xEcppo)

**Comment by sanjaiganesh** [comment]
> {

same feedaback as in 2019. (except the breaking change related comments)

---
Refers to: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json:1 in 6c15925. [](commit_id = 6c159256e9b5a3763d45af4cf94de631db4b9112, deletion_comment = False)

File: specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json (hunk 3 above)
Comment (rileymckenna) [reply specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json]: This is a child resource of Peering. Do we still need to add this?

**Review by majastrz — DISMISSED** [review]
Signed off from ARM side.

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/peerings/registeredAsns": {

This is also an invalid LIST path.

---
Refers to: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json:697 in e3182e5. [](commit_id = e3182e5e2778222b24bf05f0562e20ffb615de4f, deletion_comment = False)

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Peering/peerings/registeredPrefixes": {

This is also not a valid LIST path.

---
Refers to: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json:971 in e3182e5. [](commit_id = e3182e5e2778222b24bf05f0562e20ffb615de4f, deletion_comment = False)

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/peerings/registeredPrefixes": {

And here as well.

---
Refers to: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json:1017 in e3182e5. [](commit_id = e3182e5e2778222b24bf05f0562e20ffb615de4f, deletion_comment = False)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side (again :))

File: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json [hunk 9]
```diff
@@ -0,0 +1,3271 @@
...
+      }
+    }
+  },
+  "definitions": {
+    "CheckServiceProviderAvailabilityInput": {
+      "description": "Class for CheckServiceProviderAvailabilityInput",
+      "type": "object",
+      "properties": {
+        "peeringServiceLocation": {
+          "description": "Gets or sets the PeeringServiceLocation",
```
Comment (MyronFanQiu) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json]: what's meaning of the `Gets or sets`?

File: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json [hunk 10]
```diff
@@ -0,0 +1,3271 @@
...
+          "type": "string"
+        },
+        "peeringServiceProvider": {
+          "description": "Gets or sets the PeeringServiceProvider",
```
Comment (MyronFanQiu) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json]: same here

File: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json [hunk 11]
```diff
@@ -0,0 +1,3271 @@
...
+          "type": "integer"
+        },
+        "provisionedBandwidthInMbps": {
+          "format": "int32",
+          "description": "The bandwidth that is actually provisioned.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "sessionAddressProvider": {
+          "description": "The
```
Comment (MyronFanQiu) [inline specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json]: I don't find the usage of this definition. Do we need this one?

**Review by mmyyrroonn — APPROVED** [review]
a few comments. LGTM in general

File: specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json (hunk 11 above)
Comment (rileymckenna) [reply specification/peering/resource-manager/Microsoft.Peering/preview/2020-01-01-preview/peering.json]: removed.

**Comment by MyronFanQiu** [comment]
@NullMDR Hi. I tried twice and based on the log, looks like we meet an error for .Net automation. Could you please check?

**Pull request merged**